src/draggable-list/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { DNode } from '@dojo/framework/core/interfaces';
+import drag, { Position } from '@dojo/framework/core/middleware/drag';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import dimensions from '@dojo/framework/core/middleware/dimensions';
+import { ThemeProperties } from '@dojo/framework/core/middleware/theme';
+
+import theme from '../middleware/theme';
+import * as css from '../theme/default/draggable-list.m.css';
+import { uuid } from '@dojo/framework/core/util';
+import GlobalEvent from '../global-event';
+
+export function move<T = any>(items: T[], from: number, to: number): T[] {
```

Comment:
I don't think we should be exporting util functions from our widgets like this, nor would this be required if the widget returned the new sort order in the `onValue` callback.

Reply:
I think in most real application cases, the `from` and `to` index will be much more useful than a shallow-copied array, since this will most likely be used to hit an API or to do other internal data manipulation. I think we Dojo should absolutely export an `arrayMove` function somewhere since it's tedious to write in a type-safe manner, even if we decide to return `from`, `to`, and a shallow copy of `items` via `onMove`.